p2panda-discovery-next/src/tests.rs

```diff
@@ -249,3 +251,129 @@ async fn naive_protocol() {
 
     handle.await;
 }
+
+#[tokio::test]
+async fn topic_discovery() {
```

Comment:
I would maybe move these tests into `psi_hash` as on lib-level `tests` we're usually handling E2E tests spanning multiple systems (random walker, discovery protocol, etc.).